### DISCUSSION TITLE
Capturing output from programs in bash shell doesn't work

### DISCUSSION THREAD

**Discussion by weegreenblobbie** [community/community#27131] [Actions]
<p>I’m trying to process a code coverage file produced by lcov, and trying to pipe the output through some bash commands:</p>
<pre><code class="lang-auto">    run: |
        lines=`lcov --summary coverage.info | grep -Eo ' [0-9.]+%' | xargs | awk '{print $1}' || echo 'error'`
        echo "lines = ${lines}"
</code></pre>
<p>However the output isn’t captured, and this is what is printed instead:</p>
<pre><code class="lang-auto">    Summary coverage rate:
      lines......: 87.6% (903 of 1031 lines)
      functions..: 96.1% (199 of 207 functions)
      branches...: no data found
    lines =
    error
</code></pre>
<p>Running the command locally works as expected:</p>
<pre><code class="lang-auto">    lcov --summary coverage.info | grep -Eo ' [0-9.]+%' | xargs | awk '{print $1}' || 'error'
    87.4%
</code></pre>
<p>Anybody know how to make this work?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I guess that for some reason (e.g. different version, platform, or build options) the <code>lcov</code> version on the runner writes to stderr instead of stdout, so the regular pipe doesn’t capture the output.</p>
<p>The easiest solution should be to use <code>|&amp;</code> to capture both stdout and stderr:</p>
<pre><code class="lang-yaml">    run: |
      lines=`lcov --summary coverage.info |&amp; grep -Eo ' [0-9.]+%' | xargs | awk '{print $1}' || echo 'error'`
      echo "lines = ${lines}"
</code></pre>

**Reply by ervanipank05** [reply]
lcov

**Comment by weegreenblobbie** [comment]
<p>It’s not obvious, but even processing simple commands like:</p>
<pre><code class="lang-auto">    my_var=$(echo "one two three" |&amp; grep -Eo ' (two) ' | xargs || true)
    echo "my_var = '${my_var}'"
</code></pre>
<p>Requires using <code>|&amp;</code>.</p>
